File: c++/1122.cpp
//
//  main.cpp
//  test
//
//  Created by 李寻欢 on 2019/10/15.
//  Copyright © 2019 李寻欢. All rights reserved.
//
#include <stdio.h>
#include <iostream>
#include <string>
#include <map>
#include <cmath>
#include <algorithm>
#include <stack>
#include <vector>
#include <queue>
#include <unordered_map>
#include <unordered_set>
#include <string.h>
#include <set>
#include <deque>
using namespace std;

const int MAXN = 210;
int n,m,k;
bool g[MAXN][MAXN];

int main(){
    fill(g[0], g[0]+MAXN*MAXN, false);
    scanf("%d%d",&n,&m);
    int a,b;
    for (int i=0; i<m; ++i) {
        scanf("%d%d",&a,&b);
        g[a][b] = g[b][a] = true;
    }
    int querys;
    scanf("%d",&querys);
    for(int i=0;i<querys;++i){
        scanf("%d",&k);
        vector<int> v(k);
        set<int> s;
        for(int j=0;j<k;++j){
            scanf("%d",&v[j]);
            s.insert(v[j]);
        }
        if(s.size() != n || k-1 != n || v[0]!=v[k-1]) printf("NO\n");
        else{
            bool flag = true;
            for(int j=0;j<n;++j){
                if(g[v[j]][v[j+1]] == false){
                    printf("NO\n");
                    flag = false;
                    break;
                }
            }
            if(flag)
                printf("YES\n");
        }
        
    }
    
}

File: Python3/1122.py
n,m = map(int,input().split())
g = [[False for i in range(n+1)] for j in range(n+1)]
for _ in range(m):
    a,b=map(int,input().split())
    g[a][b] = g[b][a] = True
k = int(input())
for _ in range(k):
    v,*path = map(int,input().split())
    s = set(path)
    if path[0] != path[-1] or v != n+1 or len(s) != n:
        print("NO")
        continue
    flag = True
    for i in range(v-1):
        if g[path[i]][path[i+1]] == False:
            flag = False
            break
    if flag:
        print("YES")
    else:
        print("NO")
